wallet/src/de/schildbach/wallet/service/WalletTransactionMetadataProvider.kt

```diff
@@ -207,55 +207,57 @@ class WalletTransactionMetadataProvider @Inject constructor(
         var addressMetadata: AddressMetadata? = null
         val tx = walletData.wallet!!.getTransaction(txId)
         tx?.run {
-            if (tx.getValue(walletData.wallet!!).isNegative) {
-                // outgoing transaction, check inputs
-                for (input in inputs) {
-                    if (input.connectedOutput != null) {
-                        val output = input.connectedOutput!!
-                        val address = when {
-                            ScriptPattern.isP2PKH(output.scriptPubKey) ->
-                                Address.fromPubKeyHash(
-                                    walletData.networkParameters,
-                                    ScriptPattern.extractHashFromP2PKH(output.scriptPubKey)
-                                )
-                            else -> null // there shouldn't be other output types on our tx's
-                        }
-                        if (address != null) {
-                            val metadata =
-                                addressMetadataDao.loadSender(address.toBase58())
-                            if (metadata != null) {
-                                addressMetadata = metadata
-                                break
-                            }
-                        }
-                    }
-                }
-            } else {
-                // incoming transaction, checkout puts
-                for (output in outputs) {
+            // outgoing transaction, check inputs
+            for (input in inputs) {
+                if (input.connectedOutput != null) {
+                    val output = input.connectedOutput!!
```

Comment:
The main change here is removing the check to see if the balance is negative.

For coinable we mark the destination address as TransferOut and the transaction value is negative.  The original code was only checking inputs for addresses in the database.  However, for this case, we should be checking for the outputs in the database.

So this modified code will first check inputs, then outputs and it will return on the first match.